pkg/controller/mpi_job_controller.go

```diff
@@ -691,6 +692,10 @@ func (c *MPIJobController) getOrCreatePodGroups(mpiJob *kubeflow.MPIJob) (metav1
 		c.recorder.Event(mpiJob, corev1.EventTypeWarning, ErrResourceExists, msg)
 		return nil, fmt.Errorf(msg)
 	}
+
+	if updatedPodGroup, equal := c.podGroupCtrl.pgSpecsAreEqual(podGroup, newPodGroup); !equal {
```

Comment:
oh, I see what you did and now I understand why the previous implementation was necessary: you can't update the spec unless you know the concrete type.

Sorry for the back and forth, but I think the previous implementation was better

Reply:
I'm ok. Let's revert this commit.